atlasdb-cassandra/src/test/java/com/palantir/atlasdb/cassandra/backup/TransactionAborterTest.java

```diff
@@ -0,0 +1,307 @@
+    public void willNotAbortTimestampsOutsideRange() {
+        ImmutableList<TransactionTableEntry> rows = ImmutableList.of(
+                TransactionTableEntries.committedLegacy(BACKUP_TIMESTAMP + 10, BACKUP_TIMESTAMP + 12),
+                TransactionTableEntries.committedTwoPhase(
+                        BACKUP_TIMESTAMP + 10, PutUnlessExistsValue.committed(BACKUP_TIMESTAMP + 13)),
+                TransactionTableEntries.committedLegacy(BACKUP_TIMESTAMP + 10, BACKUP_TIMESTAMP + 14));
+        setupAbortTimestampTask(rows, FullyBoundedTimestampRange.of(Range.closed(BACKUP_TIMESTAMP + 15, 1000L)));
+
+        Stream<TransactionTableEntry> transactionsToAbort = transactionAborter.getTransactionsToAbort(
+                KEYSPACE, transactionInteraction, tableMetadata, BACKUP_TIMESTAMP);
+        assertThat(transactionsToAbort.count()).isZero();
+    }
+
+    @Test
+    public void willNotAbortTimestampsAlreadyAborted() {
+        ImmutableList<TransactionTableEntry> rows =
+                ImmutableList.of(TransactionTableEntries.explicitlyAborted(BACKUP_TIMESTAMP + 1));
+        setupAbortTimestampTask(rows, TIMESTAMP_RANGE);
+
+        Stream<TransactionTableEntry> transactionsToAbort = transactionAborter.getTransactionsToAbort(
+                KEYSPACE, transactionInteraction, tableMetadata, BACKUP_TIMESTAMP);
+        assertThat(transactionsToAbort.count()).isZero();
+    }
+
+    @Test
+    public void willAbortTimestampInRange() {
+        ImmutableList<TransactionTableEntry> rows = ImmutableList.of(
+                TransactionTableEntries.committedLegacy(BACKUP_TIMESTAMP + 1, BACKUP_TIMESTAMP + 2),
+                TransactionTableEntries.committedTwoPhase(
+                        BACKUP_TIMESTAMP + 3, PutUnlessExistsValue.committed(BACKUP_TIMESTAMP + 4)));
+        setupAbortTimestampTask(rows, FullyBoundedTimestampRange.of(Range.closed(25L, 1000L)));
+
+        Stream<TransactionTableEntry> transactionsToAbort = transactionAborter.getTransactionsToAbort(
+                KEYSPACE, transactionInteraction, tableMetadata, BACKUP_TIMESTAMP);
+        assertThat(transactionsToAbort.count()).isEqualTo(2L);
+    }
+
+    @Test
+    public void willAbortTimestampsHigherThanInputTimestamp() {
+        ImmutableList<TransactionTableEntry> rows = ImmutableList.of(
+                TransactionTableEntries.committedLegacy(BACKUP_TIMESTAMP + 1, BACKUP_TIMESTAMP + 2),
+                TransactionTableEntries.committedTwoPhase(
+                        BACKUP_TIMESTAMP + 3, PutUnlessExistsValue.committed(BACKUP_TIMESTAMP + 4)));
+        setupAbortTimestampTask(rows, TIMESTAMP_RANGE);
+
+        Stream<TransactionTableEntry> transactionsToAbort = transactionAborter.getTransactionsToAbort(
+                KEYSPACE, transactionInteraction, tableMetadata, BACKUP_TIMESTAMP);
+        assertThat(transactionsToAbort.count()).isEqualTo(2L);
+    }
+
+    @Test
+    public void willAbortTimestampsThatStartBeforeButEndAfterInputTimestamp() {
```

Comment:
How about timestamps that start after beckup but end outside of range?

Reply:
it will abort those, but not transactions that start and end after the range.